rel/parse_filterql.go

```diff
@@ -39,13 +40,22 @@ func (f *FilterQLParser) FuncResolver(funcs expr.FuncResolver) *FilterQLParser {
 }
 
 func (f *FilterQLParser) setLexer(statement string) {
-	l := lex.NewFilterQLLexer(statement)
+	l := lex.NewLexerWithOptions(lex.WithInput(statement),
+		lex.WithDialect(lex.FilterQLDialect),
+		lex.WithInitialState(lex.LexDialectForStatement),
+		lex.WithDepthLimit(f.lexerDepthLimit),
+	)
 	f.l = l
 	f.fs = nil
 	f.comment = ""
 	f.filterTokenPager = newFilterTokenPager(l)
 }
 
+// SetOverrideLexerDepth sets whether to override lexer's depth
+func (f *FilterQLParser) SetOverrideLexerDepth(dl int) {
+	f.lexerDepthLimit = dl
+}
```

Comment:
[q] is this used?

Reply:
yeah this will be used when calling from LIO, to override the lexer depth limit of parsing nested expression. We disable that check when anyone sets it to -ve number. By default, it will be 1000 as it is right now.

Reply:
@sumankarki What's `-ve number`?

Reply:
any negative number

Reply:
OH I seem, `-ve` means negative, I didn't get it at first.